### ISSUE TITLE
one toy, one room enhancement ideas for paper

### ISSUE THREAD

**Issue by nhaber** [neuroailab/curiosity_deprecated#2]
# nice things for paper!

## more robust simulation

different physx objects
flex simulation
do RGB instead of depth

## experiments and controls

does the encoder get better at the offline action postdiction 
	carefully controlled offline action postdiction (controlled object-there ratios)
does it learn to get better at object localization and presence?
does the encoder learn something better about object recognition (gabor filters?)
compare curiosity with random policy
different train and test objects

- 	test ego motion vs one object
- 	one object vs another

multi-room problem

- 	can it do ego motion in multi-room environment, does it all carry over?
- 	not "can it do slam"

## ablation
one timestep ahead
many more timesteps ahead k_f = 10
action postdiction info given
x compare with q learning assuming want to maximize (discounted) sum of losses over time

- 	or deterministic policy gradients etc

loss minimizer

# Other ideas

## degeneracy map stuff

generic solution
future prediction-action position pairing *just so*

- 	use latent space (determined by action postdiction) future prediction, curiosity signal = weighted sum of future and action

getting towards this: understanding the degeneracy and task, getting upper and lower bounds on object-there performance given certain assumptions about policy

policy hack around problem

hyperparameter choices that make this problem better than 60-45 gap

## change parametrization of action

include position, learn where it pokes

## interesting observables

characterize

- 	aside from ego control, how does it play?
- 	differences across objects

cause

- 	use different policy choices to get more interesting behavior

**Comment by nhaber** [comment]
one block object
bring fully online w freezing
compare curiosity with random policy


different physx objects
does the encoder learn something better about object recognition (gabor filters?)


does it learn to get better at object localization and presence?



different train and test objects
- 	test ego motion vs one object (train with ego motion and test with an object)
- 	one object vs another


ablations 

one timestep ahead
many more timesteps ahead k_f = 10
how much info one gives the action model (number of frames, previous actions...)

**Comment by nhaber** [comment]
Experiments

Seeding to understand variance without changing anything


A bunch of different objects, one run per object


A bunch of different objects, 20 objects on different environments with one run


A bunch of objects at once
    


static objects and a moving one
    tables and not

**Comment by nhaber** [comment]
Figures

3. a. Testing loss on action prediction problem. Bars for speed.
	b. Ablations


(or a separate figure for ablations)

4.a Object is there or not losses.
   b. Performance on 20-object classification.

Discussion

5. a. Show uncertainty maps and uncertainty map loss traces.
    b. Uncertainty map output, conditioned on object there or not
    c. object frequency
    d. action distribution -- characterize action distributions at different timepoints.
